### PR TITLE
docs: update and shorten server example

### PR DESCRIPTION
**Pull request by iiio2**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Here the single line ```export default defineEventHandler((event) => `Hello, ${event.context.params.name}!`)``` is long. That is why the full code becomes blur in the doc. 
blur
It is not readable and we cannot see the full code by default in the doc. Making it two lines removes the blur issue.

I think it is required because by default the line becomes blur in the doc. we cannot see the entire line because of long single line of code. That is why we can make it readable.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.guide/2.directory-structure/1.server.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 ---
 navigation.icon: IconDirectory
 title: server
-head.title: 'server/'
+head.title: "server/"
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.server.md:4]: would you be able to revert the formatting changes in the rest of the file? 🙏

Comment (iiio2) [reply docs/2.guide/2.directory-structure/1.server.md]: done.

File: docs/2.guide/2.directory-structure/1.server.md [hunk 2]
```diff
@@ -152,7 +152,10 @@ Server routes can use dynamic parameters within brackets in the file name like `
 **Example:**
 
 ```ts [server/api/hello/[name\\].ts]
-export default defineEventHandler((event) => `Hello, ${event.context.params.name}!`)
+export default defineEventHandler((event) => {
+  const { name } = event.context.params
+  return { `Hello, ${name}!` }
+}
```
Comment (HigherOrderLogic) [inline docs/2.guide/2.directory-structure/1.server.md:158]:
```suggestion
})
```

File: docs/2.guide/2.directory-structure/1.server.md [hunk 3]
```diff
@@ -152,7 +152,10 @@ Server routes can use dynamic parameters within brackets in the file name like `
 **Example:**
 
 ```ts [server/api/hello/[name\\].ts]
-export default defineEventHandler((event) => `Hello, ${event.context.params.name}!`)
+export default defineEventHandler((event) => {
+  const { name } = event.context.params
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.server.md:156]:
```suggestion
  const name = getRouterParam(event, 'name')
```

File: docs/2.guide/2.directory-structure/1.server.md [hunk 4]
```diff
@@ -152,7 +152,10 @@ Server routes can use dynamic parameters within brackets in the file name like `
...
+  return { `Hello, ${name}!` }
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.server.md:157]:
```suggestion
  return `Hello, ${name}!`
```

**Pull request merged**